### PR TITLE
test: e2e tests for manage account page

### PR DESCRIPTION
**Pull request by olgapod**
## Description

- Added e2e tests for the Manage Account page. 
- Added a package.json script to run the app in the preview mode  (using `vite preview`) 

## Testing

* Step 1. Spin up the dev environment and seed the test data (`npm run dev ` and then `node scripts/create-data.js`)
* Step 2. Remove your local playwright trace files `rm -rf playwright-report/` (from the root of the project )
* Step 3. Run `npm run preview` to build and start the frontend in preview mode
* Step 4. Run the tests in a separate terminal `npm run e2e`
* Step 5. The tests should pass ( if any failures - please send me .zip files from /playwright-report/data directory)

## TODO

- [ ] Write tests to cover changing the displayName  value in ManageAccount. (Need a reliable method to simulate the 90-day restriction on edits for this field)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "typecheck": "tsc --noEmit",
     "frontend": "ulimit -n 2048 && vite",
     "frontend:safe": "SAFE_ENABLED=true ulimit -n 2048 && vite",
+    "preview": "ulimit -n 2048 && NODE_ENV=development npm run build && npx import-meta-env -x .env && vite preview --port=9091",
```
Comment (rdig) [inline package.json:28]:
```suggestion
    "preview": "NODE_ENV=development npm run build && npx import-meta-env -x .env && vite preview --port=9091",
```

**Review by rdig — APPROVED** [review]
The preview logic works well, it just doesn't need the `ulimit` as you don't have as many files needed to be watched by `vite` anymore _(since they're bundled now)_

However I do want to note that while this is very close to the production build, it's still not 100% the same as production, as due to internal routing and cors issues we're running in our infrastructure, locally this cannot function exactly the same.

But this is as close as it gets, so I think it'll be of great help.

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice job on this @olgapod ! All tests are passing 🎉 

Screenshot 2024-11-22 at 11 02 20

I've left a few small comments, some are personal opinion, I think removing the unneeded awaits in the spec file is the only one that should be fixed before merging.

Another personal opinion thing, but it feels like the spec files end up being quite long and these are some of the simpler sections of the site, so it will only get worse with more complicated sections. I'm not sure if it's possible to break them up into smaller chunks?

It might also be worth adding extra tests to the avatar upload that the file type restrictions are respected, and the minimum resolution.

File: src/components/frame/v5/pages/UserAdvancedPage/partials/NotificationsSection/NotificationsSection.tsx [hunk 2]
```diff
@@ -40,18 +40,22 @@ const MSG = defineMessages({
...
-      toastTextDisabled={formatText(MSG.toastNotificationsDisabled)}
-    />
-  );
-
   return (
-    <SettingsRow.Container className="border-none">
+    <SettingsRow.Container
+      className="border-none"
+      testId="notifications-section"
+    >
```
[the remark this answers is not in the document]
Comment (olgapod) [reply src/components/frame/v5/pages/UserAdvancedPage/partials/NotificationsSection/NotificationsSection.tsx:58]: Thanks for spotting this. Performance-wise, isn’t it better to keep such things outside ? The toggleButton has no dependency on anything from inside the NotificationsSection , so why reassign it on every render? But if it isn’t a concern, and looks neater to the team I’m happy to move it back.

File: playwright/e2e/manage-account.spec.ts [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+      ).toBeVisible();
+
+      await expect(
+        page.getByText(/You can change your username every 90 days/i),
+      ).toBeVisible();
+
+      const displayNameInput = page.locator('[name="displayName"]');
+
+      // Should be pre-filled with the user's display name
+      await expect(displayNameInput.inputValue).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (olgapod) [reply playwright/e2e/manage-account.spec.ts:56]: hey, thanks a lot for this one, although this is not really the case that the await is not needed here, but I should have called it as it's a function returning promise. 
image

**Comment by olgapod** [comment]
> Another personal opinion thing, but it feels like the spec files end up being quite long and these are some of the simpler sections of the site, so it will only get worse with more complicated sections. I'm not sure if it's possible to break them up into smaller chunks?

> It might also be worth adding extra tests to the avatar upload that the file type restrictions are respected, and the minimum resolution.

Hey @iamsamgibbs , thanks for the thorough review!

Regarding the test files being lengthy - these two actually are likely to be the largest ones due to the features being a multi-tab page and a multi-step form. I'll aim to keep upcoming tests shorter moving forward.

As for adding tests for the missing cases of file uploader validation, I will definitely do that, thanks for noticig. However, I was also thinking whether adding detailed e2e tests for input validations would be the best approach, I don't mean this particular case but in general. It might be more effective to cover such validation in unit tests since they run much faster.  What are your thoughts?

**Review by Nortsova — DISMISSED** [review]
Thank you @olgapod for your work, it seems that tests are passed successfully 🎉 

image

**Review by iamsamgibbs — APPROVED** [review]
Thanks for the changes! All the tests are still passing.

Screenshot 2024-11-29 at 11 33 11

VSCode is still bugging me about this, but happy to leave it as everything is working fine.

Screenshot 2024-11-29 at 11 25 03

Also still think it would be beneficial to add the extra tests to the avatar upload that the file type restrictions are respected, and the minimum resolution. But this can always come in a later PR.

**Review by Nortsova — APPROVED** [review]
Thanks, tests are passing:

image
